### PR TITLE
[lit] [SPIRV] convert layout related tests to lit FileCheck test

### PR DESCRIPTION
**Pull request by python3kgae**
setScalar/GL/DxLayout will be replaced by -fvk_use_scalar/gl/dx_layout to match the validation behavior in runFileTest.

vk.layout.shader-record-ext.std430.hlsl and vk.layout.shader-record-nv.std430 fails after add -fvk-use-gl-layout.
Updated the CHECK line to match the new output with -fvk-use-gl-layout.

### CODE REVIEW

File: tools/clang/test/CodeGenSPIRV_Lit/vk.layout.asbuffer.std430.hlsl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// RUN: %dxc -T vs_6_0 -E main
+// RUN: %dxc -T vs_6_0 -E main -fvk-use-gl-layout -fcgl  %s -spirv | FileCheck %s
```
Comment (Keenuts) [inline tools/clang/test/CodeGenSPIRV_Lit/vk.layout.asbuffer.std430.hlsl:1]: This one passes on my end without the new layout option if I clone the PR and run lit:
`./build/bin/llvm-lit tools/clang/test/ --filter=CodeGenSPIRV_Lit/vk.layout.asbuffer -v`
(fetched your PR, so we should have the same base)

File: tools/clang/test/CodeGenSPIRV_Lit/vk.layout.cbuffer.nested.empty.std140.hlsl [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// RUN: %dxc -T vs_6_0 -E main
+// RUN: %dxc -T vs_6_0 -E main -fvk-use-gl-layout -fcgl  %s -spirv | FileCheck %s
```
Comment (Keenuts) [inline tools/clang/test/CodeGenSPIRV_Lit/vk.layout.cbuffer.nested.empty.std140.hlsl:1]: same here, passes without the new flag.

File: tools/clang/test/CodeGenSPIRV_Lit/vk.layout.cbuffer.nested.std140.hlsl [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// RUN: %dxc -T vs_6_0 -E main
+// RUN: %dxc -T vs_6_0 -E main -fvk-use-gl-layout -fcgl  %s -spirv | FileCheck %s
```
Comment (Keenuts) [inline tools/clang/test/CodeGenSPIRV_Lit/vk.layout.cbuffer.nested.std140.hlsl:1]: same here, passes without the new flag.

File: tools/clang/test/CodeGenSPIRV_Lit/vk.layout.sbuffer.nested.std430.hlsl [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// RUN: %dxc -T ps_6_0 -E main
+// RUN: %dxc -T ps_6_0 -E main -fvk-use-gl-layout -fcgl  %s -spirv | FileCheck %s
```
Comment (Keenuts) [inline tools/clang/test/CodeGenSPIRV_Lit/vk.layout.sbuffer.nested.std430.hlsl:1]: same here, passes without layout

File: tools/clang/test/CodeGenSPIRV_Lit/vk.layout.csbuffer.std430.hlsl [hunk 5]
```diff
@@ -1,4 +1,4 @@
-// RUN: %dxc -T vs_6_0 -E main
+// RUN: %dxc -T vs_6_0 -E main -fvk-use-gl-layout -fcgl  %s -spirv | FileCheck %s
```
Comment (Keenuts) [inline tools/clang/test/CodeGenSPIRV_Lit/vk.layout.csbuffer.std430.hlsl:1]: same here, passes without layout

File: tools/clang/test/CodeGenSPIRV_Lit/vk.layout.shader-record-ext.std430.hlsl [hunk 6]
```diff
@@ -1,4 +1,4 @@
-// RUN: %dxc -T lib_6_3 -fspv-target-env=vulkan1.2
+// RUN: %dxc -T lib_6_3 -fspv-target-env=vulkan1.2 -fvk-use-gl-layout -fcgl  %s -spirv | FileCheck %s
```
Comment (Keenuts) [inline tools/clang/test/CodeGenSPIRV_Lit/vk.layout.shader-record-ext.std430.hlsl:1]: This one passes as-is, with no new flag, and the same old offsets.

File: tools/clang/test/CodeGenSPIRV_Lit/vk.layout.shader-record-nv.std430.hlsl [hunk 7]
```diff
@@ -1,4 +1,4 @@
-// RUN: %dxc -T lib_6_3 -fspv-extension=SPV_NV_ray_tracing
+// RUN: %dxc -T lib_6_3 -fspv-extension=SPV_NV_ray_tracing -fvk-use-gl-layout -fcgl  %s -spirv | FileCheck %s
```
Comment (Keenuts) [inline tools/clang/test/CodeGenSPIRV_Lit/vk.layout.shader-record-nv.std430.hlsl:1]: same here, passes on my end with no modifications.

File: tools/clang/test/CodeGenSPIRV_Lit/vk.layout.tbuffer.std430.hlsl [hunk 8]
```diff
@@ -1,4 +1,4 @@
-// RUN: %dxc -T vs_6_0 -E main
+// RUN: %dxc -T vs_6_0 -E main -fvk-use-gl-layout -fcgl  %s -spirv | FileCheck %s
```
Comment (Keenuts) [inline tools/clang/test/CodeGenSPIRV_Lit/vk.layout.tbuffer.std430.hlsl:1]: same here

File: tools/clang/test/CodeGenSPIRV_Lit/vk.layout.texture-buffer.std430.hlsl [hunk 9]
```diff
@@ -1,4 +1,4 @@
-// RUN: %dxc -T vs_6_0 -E main
+// RUN: %dxc -T vs_6_0 -E main -fvk-use-gl-layout -fcgl  %s -spirv | FileCheck %s
```
Comment (Keenuts) [inline tools/clang/test/CodeGenSPIRV_Lit/vk.layout.texture-buffer.std430.hlsl:1]: same here

**Review by Keenuts — COMMENTED** [review]
Patched your PR, and I can run those lit tests just fine with no new flags or modified offsets.
Do you have repro steps?

**Comment by python3kgae** [comment]
> Patched your PR, and I can run those lit tests just fine with no new flags or modified offsets. Do you have repro steps?

I added new flags to match the setGL/Dx/ScalarLayout in the original test which affect validation in runFileTest.
I can remove the added flags if that's not required.

**Comment by s-perron** [comment]
> > Patched your PR, and I can run those lit tests just fine with no new flags or modified offsets. Do you have repro steps?
> 
> I added new flags to match the setGL/Dx/ScalarLayout in the original test which affect validation in runFileTest. I can remove the added flags if that's not required.

I would leave the flags to match what they did before. If we decide we want to remove the flags, then we will do that in a separate PR.

**Comment by Keenuts** [comment]
So adding the flags in the end might be the correct behavior, since we want to test the layout (but it does change the test)

**Review by Keenuts — APPROVED** [review]

**Review by s-perron — APPROVED** [review]

**Pull request merged**